### ISSUE TITLE
Microphone access crashes with sandbox enabled in OSX

### ISSUE THREAD

**Issue by ajrdileva** [AudioKit/AudioKit#1256]
AKSettings.audioInputEnabled = true
        mic = AKMicrophone()
        if let inputs = AudioKit.inputDevices {
            try! AudioKit.setInputDevice(inputs[0])
            try! mic.setDevice(inputs[0])
        }
        
        tracker = AKFrequencyTracker(mic)
        silence = AKBooster.init(tracker, gain: 0)
       Audiokit.output = silence

Runs perfectly with sandbox disabled but as soon as I enable it it gives me an error and crashes

ERROR:    >avae> AVAudioEngine.mm:275: AttachNode: required condition is false: node != nil
An uncaught exception was raised
required condition is false: node != nil

**Comment by aure** [comment]
How about just not enabling sandbox?

**Comment by ajrdileva** [comment]
Is it possible to submit an OS X app to the Mac App Store without enabling
sandbox?

**Comment by aure** [comment]
I don't know, but this seems less to be an issue with AudioKit and more a question for StackOverflow.

**Comment by pshannon1000** [comment]
Still struggling with this issue, as it is not possible to submit an app to the Mac App Store without enabling sandbox...

Adding the com.apple.security.temporary-exception.audio-unit-host entitlement does theoretically solve the problem, if the user agrees to a pop-up dialog from Apple with this text: 

"Use of the requested Audio Unit(s) require lowering the security settings for [App Name]. Are you sure you want to proceed?"

Most users are not going to agree to that...

If they don't, AudioKit crashes with the error when try AudioKit.start() is called:
*** Terminating app due to uncaught exception 'com.apple.coreaudio.avfaudio', reason: 'required condition is false: node != nil'


There doesn't seem to be any way to resolve the problem other than to make AudioKit sandbox safe?
- It does seem to me like this could be an issue for this forum, because the only way to resolve this issue seems to be from within the AudioKit framework.  

- I've been trying to edit the AudioKit framework to make it sandbox safe, but I just don't know the code well enough.  It seems like other users must be encountering this problem - anyone trying to use the app in an app for the Mac App Store?  Any help would be appreciated.

**Comment by aure** [comment]
I'll reopen this issue for you @pshannon1000 maybe someone will help.

**Comment by relpot** [comment]
I'm also working on an app for Mac using AudioKit and I had the same issue with AudioKit and Mac SandBox and this dirty security popup alert.

After searching in AudioKit lib how to remove this security popup alert,
I have found a fix by changing the AKComponent.swift class.

Indeed, in the public init(type: OSType, subType: OSType) method, after adding "AudioComponentFlags.sandboxSafe.rawValue" at the line 93, all works fine:
```
public init(type: OSType, subType: OSType) {
        self.init(componentType: type,
                  componentSubType: subType,
                  componentManufacturer: fourCC("AuKt"),
                  componentFlags: AudioComponentFlags.sandboxSafe.rawValue,
                  componentFlagsMask: 0)
    }
```

Hope it helps!

**Comment by aure** [comment]
This should be fixed as of 4.7, right @relpot ?

**Comment by relpot** [comment]
Yes I confirm, it is fixed in the 4.7!!
